src/SDK/SdkAutoloader.php

```diff
@@ -19,91 +33,164 @@
 use OpenTelemetry\SDK\Trace\SamplerFactory;
 use OpenTelemetry\SDK\Trace\SpanProcessorFactory;
 use OpenTelemetry\SDK\Trace\TracerProviderBuilder;
+use Throwable;
 
 /**
  * @psalm-suppress RedundantCast
  */
 class SdkAutoloader
 {
+    use LogsMessagesTrait;
+
     public static function autoload(): bool
     {
         if (!self::isEnabled() || self::isExcludedUrl()) {
             return false;
         }
-        Globals::registerInitializer(function (Configurator $configurator) {
-            $propagator = (new PropagatorFactory())->create();
-            if (Sdk::isDisabled()) {
-                //@see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#general-sdk-configuration
-                return $configurator->withPropagator($propagator);
-            }
-            $emitMetrics = Configuration::getBoolean(Variables::OTEL_PHP_INTERNAL_METRICS_ENABLED);
-
-            $resource = ResourceInfoFactory::defaultResource();
-            $exporter = (new ExporterFactory())->create();
-            $meterProvider = (new MeterProviderFactory())->create($resource);
-            $spanProcessor = (new SpanProcessorFactory())->create($exporter, $emitMetrics ? $meterProvider : null);
-            $tracerProvider = (new TracerProviderBuilder())
-                ->addSpanProcessor($spanProcessor)
-                ->setResource($resource)
-                ->setSampler((new SamplerFactory())->create())
-                ->build();
-
-            $loggerProvider = (new LoggerProviderFactory())->create($emitMetrics ? $meterProvider : null, $resource);
-            $eventLoggerProvider = (new EventLoggerProviderFactory())->create($loggerProvider);
-
-            ShutdownHandler::register($tracerProvider->shutdown(...));
-            ShutdownHandler::register($meterProvider->shutdown(...));
-            ShutdownHandler::register($loggerProvider->shutdown(...));
-
-            return $configurator
-                ->withTracerProvider($tracerProvider)
-                ->withMeterProvider($meterProvider)
-                ->withLoggerProvider($loggerProvider)
-                ->withEventLoggerProvider($eventLoggerProvider)
-                ->withPropagator($propagator)
-            ;
-        });
+        if (Configuration::has(Variables::OTEL_EXPERIMENTAL_CONFIG_FILE)) {
+            Globals::registerInitializer(fn ($configurator) => self::fileBasedInitializer($configurator));
```

Comment:
We'll need to disable the hook manager during initialization to avoid autoinstrumenting the SDK.

Reply:
Discussed at SIG, and we couldn't think of a scenario where SDK would be instrumented only during initialization?

Reply:
The SDK might use classes/methods that are autoinstrumented by a user (http clients etc.); SDK exports use the context that was active during the SDK initialization (related: https://github.com/open-telemetry/opentelemetry-php/pull/1304#discussion_r1599039084) -> disabling the hooks during initialization disables them for all SDK exports.

(AFAIK Java avoids this issue by loading autoinstrumentation classes in a separate classloader)

Reply:
Understood (and I now remember this happening in #878 ). I think I've done this (by making hookmanager enable/disable static methods which seemed the cleanest solution) - WDYT?

Reply:
If we want to make enabling/disabling static: methods should be moved out of `ExtensionHookManager`.
Alternatively we could provide an accessor to the global hook manager and still allow enabling / disabling hook managers independently.

Reply:
ok, re-jiggered things a little by moving the static methods into `HookManager`. How does this approach look?

The main wart I see is that I renamed the existing interface, `HookManager` to `HookManagerInterface` which looks untidy in composer.json's extra.spi. But, `HookManager::disable()` seemed like the most obvious approach.